GDJS/tests/tests/spriteruntimeobject.js

```diff
@@ -70,18 +170,180 @@ describe('gdjs.SpriteRuntimeObject', function() {
       expect(object.isCurrentAnimationName('firstAnimation')).to.be(false);
     });
 
-    it('keeps the same animation when using an invalid/empty name', function() {
+    it('keeps the same animation when using an invalid/empty name', () => {
       object.setAnimationName('unexisting animation');
       expect(object.getAnimation()).to.be(1);
       object.setAnimationName('');
       expect(object.getAnimation()).to.be(1);
     });
 
-    it('can change animation using animation index', function() {
+    it('can change animation using animation index', () => {
       object.setAnimation(2);
       expect(object.getAnimationName()).to.be('');
       object.setAnimation(0);
       expect(object.getAnimationName()).to.be('firstAnimation');
     });
   });
+
+  describe('Animation frames', () => {
+    it('should increment time elapsed frame when animation is playing', () => {
+      const runtimeGame = new gdjs.RuntimeGame({
+        variables: [],
+        // @ts-expect-error ts-migrate(2740) FIXME: Type '{ windowWidth: number; windowHeight: number;... Remove this comment to see the full error message
+        properties: { windowWidth: 800, windowHeight: 600 },
+        resources: { resources: [] },
+      });
+      const runtimeScene = new gdjs.RuntimeScene(runtimeGame);
+      runtimeScene.loadFromScene({
+        layers: [
+          {
+            name: '',
+            visibility: true,
+            effects: [],
+            cameras: [],
+
+            ambientLightColorR: 0,
+            ambientLightColorG: 0,
+            ambientLightColorB: 0,
+            isLightingLayer: false,
+            followBaseLayerCamera: true,
+          },
+        ],
+        variables: [],
+        r: 0,
+        v: 0,
+        b: 0,
+        mangledName: 'Scene1',
+        name: 'Scene1',
+        stopSoundsOnStartup: false,
+        title: '',
+        behaviorsSharedData: [],
+        objects: [],
+        instances: [],
+      });
+      const stepDurationInMilliseconds = 1000 / 60;
+      runtimeScene._timeManager.getElapsedTime = function () {
+        return stepDurationInMilliseconds;
+      };
+
+      const object = createObjectWithAnimationInScene(runtimeScene);
+      runtimeScene.addObject(object);
+
+      runtimeScene.renderAndStep(stepDurationInMilliseconds);
+
+      expect(object._frameElapsedTime).to.be(stepDurationInMilliseconds / 1000);
```

Comment:
You could even check the current frame (it was 0 before the renderAndStep, and is probably 1 or 2 now) ? As the timestep is forced, it should be stable?